File: app/graphql/types/custom/highlight_type.rb
module Types
  module Custom
    class HighlightType < Types::BaseObject
      field :id, Integer, null: false
      field :order, Integer, null: true
      field :created_at, GraphQL::Types::ISO8601DateTime, null: false
      field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
      field :article_id, Integer, null: true
      field :scheduled_time, GraphQL::Types::ISO8601DateTime, null: false
      field :published, Boolean, null: true
      field :location_id, Integer, null: false

      #Custom Fields
      field :article, resolver: Resolvers::ResolversHighlight::Fields::FieldArticle, null: false
    end
  end
end

File: angular/src/app/types/graphql/highlight-type.ts
import * as moment from "moment-timezone";
import {ArticleType} from "./article-type";

export interface HighlightType {
  id: number;
  articleId: number;
  order: number;
  scheduledTime: moment.Moment;
  published: boolean;
  locationId: number;
  createdAt: moment.Moment;
  updatedAt: moment.Moment;

  // Custom Fields
  article: ArticleType | null;
}
